src/radical/pilot/raptor/worker.py

```diff
@@ -479,13 +574,21 @@ def _dispatch(self, task):
         task['pid'] = os.getpid()
 
         # ----------------------------------------------------------------------
-        def _dispatch_thread():
+        def _dispatch_thread(tlock):
+            # FIXME: do we still need this thread?
+
             import setproctitle
-            setproctitle.setproctitle('dispatch.%s' % task['uid'])
+            setproctitle.setproctitle('rp.dispatch.%s' % task['uid'])
+
+            # make CUDA happy
+            # FIXME: assume physical device numbering for now
+            os.environ['CUDA_VISIBLE_DEVICES'] = \
```

Comment:
It is not necessary to always have `CUDA_VISIBLE_DEVICES` set, even if it is empty

Reply:
It does not hurt either though?  Anyway, shielded it now.